### ISSUE TITLE
bug when searching for a form in the database?

### ISSUE THREAD

**Issue by nataliacp** [Anaphory/lexedata#22]
Failed to find object {'cldf_languageReference': 'tocantins_asurini', 'cldf_value': '<omónem>(put on (clothes))', 'orthographic': 'omónem', 'cldf_comment': 'put on (clothes)', 'variants': []} in the database. Skipped. In cell: <Cell 'Verbs'.Z536>.

I have found a number now of these errors that seem wrong. The form in question seems to be identical and present in both spreadsheets in the correct language and source. I am pretty sure that this one cannot have been fixed in the meantime as it is in the verb tab and the few things i fixed were in the beginning.

**Comment by nataliacp** [comment]
I don't know if this is helpful, but the only common thing I have noticed in these seemingly false errors, is that the forms all have diacritics.

**Comment by nataliacp** [comment]
I have now reimported the MG dataset using the newest excel files, and I can confirm that the same items that seem correct, show up as errors.

**Comment by nataliacp** [comment]
There appears to be a remnant of this bug at least in this (admittedly crazy) form: [kʲikʷãn͡dúsu] (ciervo) {2}. it is in Siriono DEER4.5. As far as I can tell, it is identical in the two spreadsheets but it is not recognized. Also, in the error file the superscripts are not superscripts, but maybe this is a matter of plain text, right?

**Comment by Anaphory** [comment]
> Also, in the error file the superscripts are not superscripts, 
> but maybe this is a matter of plain text, right?
It is more likely an issue with normalization, and should 
definitely not happen.
I found and fixed this issue: There is four different unicode 
normalizations – NFC, NFD, NFKC, and NFKD. The Cs and the Ds mean 
‘end up with the most composed (C) or decomposed (D) string 
possible’. The ‘K’s are compatibility normalizations, and ʲ and j 
are ‘compatible’ in this sense. We used NFKC before, now I 
switched to NFC.

> As far as I can tell, it is identical in the two spreadsheets 
> but it is not recognized.
They are not the same:
>>> k1.encode('ascii', 'backslashreplace')
b'kj\\u0268kw\\xe3n\\u0361d\\xfasu'
>>> k2.encode('ascii', 'backslashreplace')
b'kjikw\\xe3n\\u0361d\\xfasu'

The form table contains a normal i, the cognate table contains ɨ 
U+0268 LATIN SMALL LETTER I WITH STROKE. It is not actually the 
same string, so a warning makes sense. We could make this match be 
found if you find more cases like this, but otherwise I don't 
think it's worth it.
